cmake/ContourThirdParties.cmake

```
@@ -124,11 +145,20 @@ else()
     set(THIRDPARTY_BUILTIN_termbench "(bench-headless disabled)")
 endif()
 
+set(BOXED_CPP_VERSION "1.2.2")
 find_package(boxed-cpp QUIET)
 if(boxed-cpp_FOUND)
     set(THIRDPARTY_BUILTIN_boxed_cpp "system package")
+    message(STATUS "Found boxed-cpp: ${boxed-cpp_VERSION} (system package)")
 else()
     ContourThirdParties_Embed_boxed_cpp()
+    subproject_version(boxed-cpp boxed_cpp_version)
+
+    # Assuming the variable is named boxed-cpp_VERSION, adjust if necessary
+    if(NOT DEFINED boxed_cpp_version OR boxed_cpp_version VERSION_LESS BOXED_CPP_VERSION)
+        message(FATAL_ERROR "Embedded boxed-cpp version must be at least ${BOXED_CPP_VERSION}, but found ${boxed_cpp_version}")
+    endif()
+
```

We need to check version if this is a system package as well
```suggestion
    subproject_version(boxed-cpp boxed-cpp_VERSION)
endif()
    # Assuming the variable is named boxed-cpp_VERSION, adjust if necessary
if(NOT DEFINED boxed-cpp_VERSION OR boxed-cpp_VERSION VERSION_LESS BOXED_CPP_MINIMAL_VERSION)
    message(FATAL_ERROR "${THIRDPARTY_BUILTIN_boxed_cpp} boxed-cpp version must be at least ${BOXED_CPP_MINIMAL_VERSION}, but found ${boxed-cpp_VERSION}")


```

---

Ok right, but we can't use `subproject_version()` for checking the version of a system installed dependency, because `subproject_version()` is checking the sub project's CMakeLists.txt the manual way to extract the version information. We don't have that for a system installed version.

If the system does not provide boxed-cpp **or** it's not a sufficient version, then we get into the else() branch and expect it to be an embedded dependency. 🤔

---

Indeed, we need to try to find version that we need and if not then check embed, so something like 
```
find_package(boxed-cpp ${BOXED_CPP_MINIMAL_VERSION} QUIET)
if(boxed-cpp_FOUND)
....
endif()
// try embeded and check version
```